packages/flutter/test/material/radio_list_tile_test.dart

```diff
@@ -2137,4 +2137,148 @@ void main() {
     expect(radioOffset.dy - tileOffset.dy, bottomPositionRadio);
     expect(secondaryOffset.dy - tileOffset.dy, bottomPositionSecondary);
   });
+
+  testWidgets('RadioListTile respects radioBackgroundColor in enabled/disabled states', (
+    WidgetTester tester,
+  ) async {
+    const Color activeEnabledBackgroundColor = Color(0xFF000001);
+    const Color activeDisabledBackgroundColor = Color(0xFF000002);
+    const Color inactiveEnabledBackgroundColor = Color(0xFF000003);
+    const Color inactiveDisabledBackgroundColor = Color(0xFF000004);
+
+    Color getBackgroundColor(Set<MaterialState> states) {
+      if (states.contains(MaterialState.disabled)) {
+        if (states.contains(MaterialState.selected)) {
+          return activeDisabledBackgroundColor;
+        }
+        return inactiveDisabledBackgroundColor;
+      }
+      if (states.contains(MaterialState.selected)) {
+        return activeEnabledBackgroundColor;
+      }
+      return inactiveEnabledBackgroundColor;
+    }
+
+    final MaterialStateProperty<Color> backgroundColor = MaterialStateColor.resolveWith(
+      getBackgroundColor,
+    );
+
+    int? groupValue = 0;
+    Widget buildApp({required bool enabled}) {
+      return wrap(
+        child: StatefulBuilder(
+          builder: (BuildContext context, StateSetter setState) {
+            return RadioGroup<int>(
+              groupValue: groupValue,
+              onChanged: (int? newValue) {
+                setState(() {
+                  groupValue = newValue;
+                });
+              },
+              child: RadioListTile<int>(
+                value: 0,
+                radioBackgroundColor: backgroundColor,
+                enabled: enabled,
+              ),
+            );
+          },
+        ),
+      );
+    }
+
+    await tester.pumpWidget(buildApp(enabled: true));
+
+    // Selected and enabled.
+    await tester.pumpAndSettle();
+    expect(
+      Material.of(tester.element(find.byType(Radio<int>))),
+      paints
```

Comment:
not something that needs to be changed, but consider checking the parameter is passed into Radio widget should be good enough as long as we have paints check in radio_test.dart.

This way we don't need to change all these test if the paint method should change.

Reply:
Good point! I'll keep that in mind for the next PRs 👍 Thank you!